src/Sylius/Component/Core/Uploader/PathGenerator.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Sylius\Component\Core\Uploader;
+
+use Gaufrette\Filesystem;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+final class PathGenerator implements PathGeneratorInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function generateForFile(Filesystem $filesystem, \SplFileInfo $file)
+    {
+        $i = 0;
+        do {
+            $generatedPath = md5(sprintf("%s.%s.%d", $file->getFilename(), $file->getMTime(), $i));
```

Comment:
I'd rather be more explicit, for the same reason I prefer not to have assignments in `if`s etc.